### PR TITLE
Refactor: slack_user_privilege_escalation

### PR DESCRIPTION
**Pull request by miotke**
### Background

When looking for an example detection I noticed that the description for slack_user_privilege_escalation was inaccurate. This PR fixes that along with making the titles more expressive. 

Also fixes a TODO item which elevates the severity level if admin/owner permissions is granted or ownership is transferred/granted to a critical severity. 

### Changes

* Fixes description of the detection 
* Makes titles more expressive passing the affected user. 
* Increases severity to critical if admin/owner is granted or ownership is transferred/grante.

### Testing

```
Slack.AuditLogs.UserPrivilegeEscalation
	[PASS] Owner Transferred
		[PASS] [rule] true
		[PASS] [title] Slack Owner Transferred from username (user@example.com)
		[PASS] [dedup] Slack Owner Transferred from username (user@example.com)
		[PASS] [alertContext] {"actor-name": "username", "actor-email": "user@example.com", "actor-ip": "1.2.3.4", "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"}
		[PASS] [severity] CRITICAL
	[PASS] Permissions Assigned
		[PASS] [rule] true
		[PASS] [title] Slack User, username (user@example.com), assigned permissions
		[PASS] [dedup] Slack User, username (user@example.com), assigned permissions
		[PASS] [alertContext] {"actor-name": "username", "actor-email": "user@example.com", "actor-ip": "1.2.3.4", "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"}
		[PASS] [severity] MEDIUM
	[PASS] Role Changed to Admin
		[PASS] [rule] true
		[PASS] [title] username (user@example.com) promoted to admin
		[PASS] [dedup] username (user@example.com) promoted to admin
		[PASS] [alertContext] {"actor-name": "username", "actor-email": "user@example.com", "actor-ip": "1.2.3.4", "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"}
		[PASS] [severity] CRITICAL
	[PASS] Role Changed to Owner
		[PASS] [rule] true
		[PASS] [title] username (user@example.com) promoted to Owner
		[PASS] [dedup] username (user@example.com) promoted to Owner
		[PASS] [alertContext] {"actor-name": "username", "actor-email": "user@example.com", "actor-ip": "1.2.3.4", "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"}
		[PASS] [severity] CRITICAL
	[PASS] User Logout
		[PASS] [rule] false

--------------------------
Panther CLI Test Summary
	Path: rules/slack_rules
	Passed: 21
	Failed: 0
	Invalid: 0
```

### CODE REVIEW

File: rules/slack_rules/slack_user_privilege_escalation.py [hunk 1]
```diff
@@ -13,16 +13,30 @@ def rule(event):
...
+    if event.get("action") == "permissions_assigned":
+        return f"Slack User, {username} ({email}), assigned permissions"
+
+    if event.get("action") == "role_change_to_admin":
+        return f"{username} ({email}) promoted to admin"
+
+    if event.get("action") == "role_change_to_owner":
+        return f"{username} ({email}) promoted to Owner"
+
     return "Slack User Privilege Escalation"
```
Comment (edyesed) [inline rules/slack_rules/slack_user_privilege_escalation.py:31]:
```suggestion
    return "Slack User Privilege Escalation event {event.get('action')} on {username} ({email})"
```

File: rules/slack_rules/slack_user_privilege_escalation.py [hunk 2]
```diff
@@ -13,16 +13,34 @@ def rule(event):
...
-    # TODO: Add case to check for admin privileges to escalate to Critical
     if event.get("action") == "permissions_assigned":
         return "Medium"
+    if event.get("action") == "role_change_to_admin":
+        return "Critical"
+    if event.get("action") == "role_change_to_owner":
+        return "Critical"
+    if event.get("action") == "owner_transferred":
+        return "Critical"
     return "High"
```
Comment (edyesed) [inline rules/slack_rules/slack_user_privilege_escalation.py:44]:
```suggestion
    return "Critical"
```

Comment (edyesed) [reply rules/slack_rules/slack_user_privilege_escalation.py:44]: There are four events in the list, and there is no way to reach line 44 for High severity

File: rules/slack_rules/slack_user_privilege_escalation.py (hunk 1 above)
Comment (edyesed) [reply rules/slack_rules/slack_user_privilege_escalation.py:31]: I like the human-friendly way of formatting the title that you've brought here. Perfectly fine to keep it as you have with a custom message for each. If you do that, Please add `Slack` as the first word to each title

Comment (miotke) [reply rules/slack_rules/slack_user_privilege_escalation.py:31]: You got it! Change incoming.

File: rules/slack_rules/slack_user_privilege_escalation.py (hunk 2 above)
Comment (miotke) [reply rules/slack_rules/slack_user_privilege_escalation.py]: Ahh this makes a good point. Everything else is Critical except for "permission_assigned". Thanks for pointing this out!

**Review by edyesed — APPROVED** [review]
updates slack user privilege escalation with helpful `title` and `description` . Also increases the severity to critical for `user becomes admin` conditions

**Pull request merged**